### PR TITLE
Prevent panic in NewSession function

### PR DESCRIPTION
**Pull request by ahmdrz**
Hi gorilla team !

### CODE REVIEW

**Comment by kisielk** [comment]
Where does this actually cause a panic?

**Comment by ahmdrz** [comment]
Here is an example :

```go
s := sessions.NewSession(store, "session-name")
s.Save(r, w)
```

On [Line 112](https://github.com/gorilla/sessions/blob/master/store.go#L112) in `store.go`

```go
// Save adds a single session to the response.
func (s *CookieStore) Save(r *http.Request, w http.ResponseWriter,
	session *Session) error {
	encoded, err := securecookie.EncodeMulti(session.Name(), session.Values,
		s.Codecs...)
	if err != nil {
		return err
	}
	http.SetCookie(w, NewCookie(session.Name(), encoded, session.Options))
	return nil
}
```

Is `session.Options` point to null , Panic occurred :

`invalid memory address or nil pointer dereference`

**Comment by kisielk** [comment]
Hm interesting that nobody noticed this before, I guess it only affects CookieStore.

Do you think you can add a test for this?

**Comment by ahmdrz** [comment]
Yes , Maybe tomorrow

On Jan 15, 2018 22:26, Kamil Kisiel <notifications@github.com> wrote:

Hm interesting that nobody noticed this before, I guess it only affects CookieStore.

Do you think you can add a test for this?

**Comment by ahmdrz** [comment]
Here is a simple test function.

**Comment by kisielk** [comment]
LGTM. @elithrar for a sanity check?

File: sessions_test.go [hunk 1]
```diff
@@ -155,6 +155,24 @@ func TestFlashes(t *testing.T) {
...
+	}
+	w := httptest.NewRecorder()
+
+	session := NewSession(store, "hello")
+
+	session.Values["data"] = "hello-world"
+
+	err = session.Save(req, w)
+	if err != nil {
+		t.Fatal("failed to save session", err)
```
Comment (elithrar) [inline sessions_test.go]: The test should "fail" in the current revision: the panic will cause the test not to run as it won't get to this stage. You should use a deferred recover() to "catch" any potential panic and *then* call `t.Fatal`.

**Comment by ahmdrz** [comment]
Hello ?

**Pull request merged**